src/commands/staff/premium.js

```diff
@@ -44,30 +44,35 @@ exports.run = async (client, message, args) => {
             .send("Thanks, <@" + userid + "> for donating $" + parser.format(amount));
     };
 
-    let userid = message.guild.members.cache.get(
-        args[2].match(/[0-9]{17,19}/).length == 0 ? args[2] : args[2].match(/[0-9]{17,19}/)[0],
-    );
-    
+    let userid = args[2].match(/[0-9]{17,19}/) ? args[2].match(/[0-9]{17,19}/)[0] : args[2];
+
     let amount = Number.parseInt(args[3]);
     if (isNaN(amount)) return;
 
-    let oldBal = await userPrem.get(userid + ".donated") || 0;
+    let userPremium = await userPrem.get(userid);
+    let oldBal = userPremium ? userPremium.donated || 0 : 0;
 
     if (args[1].toLowerCase() === "add") {
-        setDonations(userid, amount + oldBal);
+        await setDonations(userid, amount + oldBal);
         sendMessage(userid, amount);
 
-        await message.member.roles.add(Config.DiscordBot.Roles.Donator);
```

Comment:
Surely it's not supposed to always give the role to the one who ran the command?

Reply:
CC @DEV-DIBSTER not sure why you are against these changes, or at least it looks like you are

Reply:
> CC @DEV-DIBSTER not sure why you are against these changes, or at least it looks like you are - @djohts 

Not sure where you pulled that I was against such changes, considering I literally said I was going to look at this over the weekend, and that my **initial quick test** wasn't able to find anything.